core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/ObjectEntry.kt

```diff
@@ -0,0 +1,37 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.DopeQueryManager
+import ch.ergon.dope.resolvable.Resolvable
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.validtype.ObjectType
+import ch.ergon.dope.validtype.StringType
+import ch.ergon.dope.validtype.ValidType
+
+class ObjectEntry<T : ValidType>(
+    private val key: TypeExpression<StringType>,
+    private val value: TypeExpression<T>,
+) : Resolvable {
+    override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
+        val keyQuery = key.toDopeQuery(manager)
+        val valueQuery = value.toDopeQuery(manager)
+        return DopeQuery(
+            queryString = "${keyQuery.queryString} : ${valueQuery.queryString}",
+            parameters = keyQuery.parameters.merge(valueQuery.parameters),
+        )
+    }
+}
+
+fun TypeExpression<StringType>.to(value: TypeExpression<out ValidType>) = ObjectEntry(this, value)
+
+class ObjectEntryField<T : ValidType>(private val objectType: TypeExpression<ObjectType>, private val key: String) : Field<T>(key, "") {
```

Comment:
I think _objectType_ does not make sense here. An _objectEntry_ has a key and a value (that makes sense), but and _objectEntryField_ does not have a value but a type?

Reply:
I guess you should rename it to _ObjectAccess_ (similar to ArrayAccess) and can you move it an separate file?

Reply:
I renamed this class to `ObjectEntry` and moved the other class into the Primitive file and added the Suffix `Primitive` to its name.